retrieval/discovery/ec2.go

```diff
@@ -59,11 +60,19 @@ func NewEC2Discovery(conf *config.EC2SDConfig) *EC2Discovery {
 	if conf.AccessKey == "" && conf.SecretKey == "" {
 		creds = defaults.DefaultChainCredentials
 	}
+
+	var awsConfig = &aws.Config{
+		Region:      &conf.Region,
+		Credentials: creds,
+	}
+
+	client := &http.Client{Transport: &http.Transport{
```

Comment:
I would be more confident if we base this off of `http.DefaultClient` and only change `http.DefaultClient.Transport.Proxy`. That way we keep the safe default timeouts etc. that were in use by the AWS SDK before using plain `http.DefaultClient`.